qemu/tests/pvpanic_event_check.py

```diff
@@ -35,13 +36,27 @@ def run(test, params, env):
     session.cmd(setup_guest_cmd)
     session = vm.reboot(session)
     s, o = session.cmd_status_output(check_kexec_cmd)
-    if s or o is '':
+    if s or o == '':
         test.error("Failed to setup crash_kexec_post_notifiers in guest")
 
-    error_context.context("Check kdump server status in guest", test.log.info)
-    o = session.cmd_output(check_kdump_service)
-    if kdump_expect_status not in o:
-        test.error("Kdump service status is not %s" % kdump_expect_status)
+    error_context.context("Check kdump server status in guest",
+                          test.log.info)
+    max_attempts = 10
+    check_interval = 5
+    while max_attempts > 0:
+        o = session.cmd_output(check_kdump_service)
+        if o.startswith(kdump_expect_status):
+            test.log.info("kdump service reach %s status"
+                          % kdump_expect_status)
+            break
+        else:  # include o == "activating"
+            time.sleep(check_interval)
+            max_attempts -= 1
+    if max_attempts == 0:
+        test.error(
+            "Kdump service did not reach %s status "
+            "within the timeout period" % kdump_expect_status
+        )
```

Comment:
Hi @heywji 

Please try to use this function, I think this function can make your code neater。

https://github.com/avocado-framework/avocado-vt/blob/2ddf56c099e314351d624bc33d6a7289f33a9804/virttest/utils_misc.py#L535

e.g. :
utils_misc.wait_for(session.cmd_output(check_kdump_service).startswith(kdump_expect_status), 50, 0, 5)

Reply:
OK, thanks @leidwang

Reply:
Hi @leidwang, 
Could you please review this patch again? I have changed the compute method as you mentioned. 
Thanks : )

Test Result:
```
CaseName                                 Status          StartTime       EndTime         Result     TimeElapsed
01-repeat1.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:28:17 2024-02-29 13:31:38 PASS 201.324952
02-repeat2.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:31:38 2024-02-29 13:33:31 PASS 113.00829
03-repeat3.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:33:32 2024-02-29 13:35:22 PASS 110.035538
04-repeat4.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:35:22 2024-02-29 13:37:21 PASS 118.888181
05-repeat5.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:37:22 2024-02-29 13:39:20 PASS 118.257183
06-repeat6.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:39:20 2024-02-29 13:42:40 PASS 200.271852
07-repeat7.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:42:41 2024-02-29 13:44:40 PASS 119.446609
08-repeat8.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:44:41 2024-02-29 13:46:39 PASS 118.006689
09-repeat9.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:46:39 2024-02-29 13:49:14 PASS 154.859977
10-repeat10.Host_RHEL.m9.u4.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.5.0.x86_64.io-github-autotest-qemu.pvpanic_event_check.q35 Finshed 2024-02-29 13:49:14 2024-02-29 13:51:07 PASS 112.29784
Summary:
Finshed=10, PASS=10
```